Sources/SPTDataLoader/SPTDataLoaderService.m

```diff
@@ -224,6 +226,10 @@ - (NSURLSessionTask *)createTaskForRequest:(SPTDataLoaderRequest *)request
 - (void)performRequest:(SPTDataLoaderRequest *)request
 requestResponseHandler:(id<SPTDataLoaderRequestResponseHandler>)requestResponseHandler
 {
+    if (self.sessionInvalidated) {
```

Comment:
Maybe consider moving this check to `createTaskForRequest:`? Otherwise we're risking to face the same issue via the `requestTaskHandlerNeedsNewTask:` method

Reply:
I considered it but then how can `cancelledRequest` be called easily? I should change the signature of `createTaskForRequest` and return some nullable value and check for it in all call sites? To me `createTaskForRequest` is not designed to fail and when called URLSession should be valid and this is exactly the bug I am trying to fix here

Reply:
Ah I see your point now, thanks for clarifying!